modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -325,6 +325,174 @@ func (suite *KeeperTestSuite) TestChanUpgradeTry() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestChanUpgradeAck() {
+	var (
+		path                    *ibctesting.Path
+		counterpartyFlushStatus types.FlushStatus
+		counterpartyUpgrade     types.Upgrade
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
```

Comment:
I can refactor this to `expError` if people think its worth doing, can also do it in a follow up PR to this one to keep the ball moving

Reply:
I think it's good practice for us to have unit tests which make stronger assumptions

Reply:
Yeah, agree for more sensitive code like this we should aim for that! You happy if I make this refactor in a follow up PR?

Reply:
yup definitely!

Reply:
I really like the error approach, it's happened a few times where a test was failing for a different reason that the test was actually expecting. Follow up works for me